File: prime/src/main/java/org/ostelco/prime/storage/AsMappable.java
package org.ostelco.prime.storage;

import java.util.Map;

public interface AsMappable {
    /**
     * Returns a map that is intended to be written to a Firebase subtree.
     *
     * @return A map representing the content of the instance implementing
     *         this interface.
     */
    Map<String, Object> asMap();
}

File: prime/src/main/kotlin/org/ostelco/prime/storage/AsMappable.kt
package org.ostelco.prime.storage

interface AsMappable {
    /**
     * Returns a map that is intended to be written to a Firebase subtree.
     *
     * @return A map representing the content of the instance implementing
     * this interface.
     */
    fun asMap(): Map<String, Any?>
}
